docusaurus/docs/create-a-plugin/extend-a-plugin/instrument-a-backend-plugin.md

```diff
@@ -0,0 +1,421 @@
+---
+id: instrument-a-backend-plugin
+title: Instrument a backend plugin
+description: How to instrument backend plugins using logs, metrics and traces.
+keywords:
+  - grafana
+  - plugins
+  - plugin
+  - instrument
+  - instrumentation
+  - logs
+  - metrics
+  - distributed tracing
+  - tracing
+  - backend
+  - back-end
+---
+
+# Instrument a backend plugin
+
+Guidance, conventions and best practices for instrumenting backend plugins using logs, metrics and traces to make it easier to diagnose and resolve issues for both plugin developers and Grafana operators.
+
+## Logs
+
+Logs are files that record events, warnings and errors as they occur within a software environment. Most logs include contextual information, such as the time an event occurred and which user or endpoint was associated with it.
+
+### Implement logging in your plugin
+
+Using the global logger, `backend.Logger`, from the [backend package](https://pkg.go.dev/github.com/grafana/grafana-plugin-sdk-go/backend) works everywhere and for most use cases.
+
+**Example:**
+
+```go
+package plugin
+
+import (
+    "errors"
+
+    "github.com/grafana/grafana-plugin-sdk-go/backend"
+)
+
+func main() {
+    backend.Logger.Debug("Debug msg", "someID", 1)
+    backend.Logger.Info("Info msg", "queryType", "default")
+    backend.Logger.Warning("Warning msg", "key", "value")
+    backend.Logger.Error("Error msg", "error", errors.New("An error occurred"))
```

Comment:
It's clear from this code example but maybe it would be useful to include what the output of those lines is. (or at least one)

When reviewing some community plugins, I encountered some instances where developers did not realize that those are key/value pairs and output in logfmt. They were using the logger in the wrong way (logging values directly, instead of keys)

Reply:
I'm showing that under the "Inspect logs locally" section. Think that's enough or should I link to it perhaps?

Reply:
Yes I think that's great. I didn't read up to that point when I made that comment, but in this document we have plenty of examples and it should be clear to the developer what it will output :smile: